### PR TITLE
https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/360

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/java/com/efficio/fieldbook/web/label/printing/controller/LabelPrintingController.java [hunk 1]
```diff
@@ -261,16 +260,17 @@ public String showFieldmapLabelDetails(@ModelAttribute("labelPrintingForm") fina
...
 
 		model.addAttribute(LabelPrintingController.AVAILABLE_FIELDS,
-				this.labelPrintingService.getAvailableLabelFieldsForFieldMap(true, locale));
+			this.labelPrintingService.getAvailableLabelFieldsForFieldMap(true, locale));
 
 		return super.show(model);
 	}
 
 	@RequestMapping(value = "/stock/{id}", method = RequestMethod.GET)
 	public String showStockListLabelDetails(@ModelAttribute("labelPrintingForm") final LabelPrintingForm form, final Model model,
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/label/printing/controller/LabelPrintingController.java:269]: Is this still being called? **showStockListLabelDetails**

File: src/main/webapp/WEB-INF/static/js/trialmanager/manageTrial.js [hunk 2]
```diff
@@ -842,6 +707,11 @@ stockListImportNotSaved, ImportDesign, isOpenStudy, displayAdvanceList, Inventor
 							}
 						})
 					});
+					// Add crosses and selections tab after Observations tab.
+					// datasetType id 4 is plot type.
+					if (datasetType.id === 4) {
+						$scope.subObservationTabs.push($scope.crossesAndSelectionsTab);
+					}
```
Comment (nahuel-soldevilla) [inline src/main/webapp/WEB-INF/static/js/trialmanager/manageTrial.js:714]: Perhaps we can add them all after the obs-type tabs (obs and sub-obs), and keep the list in the template
```html
<li ng-repeat="tab in crossesAndSelectionsTab
```
The mockup show them after observation tab but it doesn't mention sub-obs...
Pushing them to the list of obs tabs feels wrong to me.

File: src/main/java/com/efficio/fieldbook/web/label/printing/controller/LabelPrintingController.java (hunk 1 above)
Comment (abatac) [reply src/main/java/com/efficio/fieldbook/web/label/printing/controller/LabelPrintingController.java:269]: It's not used anymore so we can delete it.

File: src/main/java/com/efficio/fieldbook/service/api/LabelPrintingService.java [hunk 3]
```diff
@@ -138,23 +120,15 @@ String generateLabels(String labelType, List<StudyTrialInstanceInfo> trialInstan
...
-	GermplasmListType getStockListType(String type);
 
 	/**
-	 * 
 	 * @param stockListId
 	 * @return
 	 * @throws MiddlewareQueryException
 	 */
-    List<InventoryDetails> getInventoryDetails(int stockListId);
+	List<InventoryDetails> getInventoryDetails(int stockListId);
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/service/api/LabelPrintingService.java:133]: `getInventoryDetails` is now unused

File: src/main/java/com/efficio/fieldbook/service/api/LabelPrintingService.java [hunk 4]
```diff
@@ -138,23 +120,15 @@ String generateLabels(String labelType, List<StudyTrialInstanceInfo> trialInstan
...
-	 * @param type
-	 * @return
-	 */
-	GermplasmListType getStockListType(String type);
 
 	/**
-	 * 
 	 * @param stockListId
 	 * @return
 	 * @throws MiddlewareQueryException
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/service/api/LabelPrintingService.java:131]: Remove @throws MiddlewareQueryException from this class

File: src/main/java/com/efficio/fieldbook/web/common/controller/CrossingSettingsController.java [hunk 5]
```diff
@@ -39,6 +39,8 @@
 import org.generationcp.middleware.pojos.presets.ProgramPreset;
 import org.generationcp.middleware.pojos.workbench.ToolName;
 import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+import org.generationcp.middleware.service.api.study.StudyGermplasmDto;
+import org.generationcp.middleware.service.api.study.StudyGermplasmService;
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/common/controller/CrossingSettingsController.java:43]: Remove unused import

File: src/main/webapp/WEB-INF/pages/TrialManager/createTrial.html [hunk 6]
```diff
@@ -807,8 +783,12 @@ <h1 class="page-header col-xs-12 edit-trial-page-identifier">
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/studyInstanceService.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/environments.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/experimentalDesign.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/saveList.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/r-package/r-package.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/visualization/visualization.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/germplasm-study-source/GermplasmStudySource.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/germplasm-study-source/GermplasmStudySourceService.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/visualization/visualization.js}"></script>
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/pages/TrialManager/createTrial.html:791]: Double import of `visualization.js`?

File: src/main/webapp/WEB-INF/pages/TrialManager/createTrial.html [hunk 7]
```diff
@@ -807,8 +783,12 @@ <h1 class="page-header col-xs-12 edit-trial-page-identifier">
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/studyInstanceService.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/environments.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/experimentalDesign.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/saveList.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/r-package/r-package.js}"></script>
 	<script type="text/javascript" th:src="@{/static/js/trialmanager/visualization/visualization.js}"></script>
+	<script type="text/javascript" th:src="@{/static/js/trialmanager/germplasm-study-source/GermplasmStudySource.js}"></script>
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/pages/TrialManager/createTrial.html:789]: Should the file names of these new JS files be camel case (ie. start with lower case)?

File: src/main/webapp/WEB-INF/static/js/trialmanager/column-filters/column-filters.html [hunk 8]
```diff
@@ -40,6 +40,11 @@
 		<input class="form-control" ng-model='filter.value' placeholder="{{filter.placeholder || 'Contains text'}}"/>
 	</study-column-filter>
 </script>
+<script type="text/ng-template" id="studyColumnFilterTextNumberOnly.html">
+	<study-column-filter>
+		<input type="number" maxlength="10" class="form-control" ng-model='filter.value' placeholder="{{filter.placeholder || 'Contains numbers'}}"/>
+	</study-column-filter>
+</script>
```
Comment (nahuel-soldevilla) [inline src/main/webapp/WEB-INF/static/js/trialmanager/column-filters/column-filters.html:47]: @abatac What do you think if we use `studyColumnFilterList` instead for numbers in _Crosses and selections_ (gid, etc)? for consistency with the inventory tab.

Comment (abatac) [reply src/main/webapp/WEB-INF/static/js/trialmanager/column-filters/column-filters.html:47]: I think it makes sense to use studyColumnFilterList for consistency across tables, we will address this during cleanup: IBP-3876

**Pull request merged**